server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -109,598 +109,510 @@ protected void masterOperation(
     ) {
         assert task instanceof CancellableTask : task + " not cancellable";
 
-        getMultipleReposSnapshotInfo(
-            request.isSingleRepositoryRequest() == false,
-            SnapshotsInProgress.get(state),
+        new GetSnapshotsOperation(
+            (CancellableTask) task,
             TransportGetRepositoriesAction.getRepositories(state, request.repositories()),
+            request.isSingleRepositoryRequest() == false,
             request.snapshots(),
             request.ignoreUnavailable(),
-            request.verbose(),
-            (CancellableTask) task,
+            SnapshotPredicates.fromRequest(request),
             request.sort(),
-            request.after(),
-            request.offset(),
-            request.size(),
             request.order(),
             request.fromSortValue(),
-            SnapshotPredicates.fromRequest(request),
-            request.includeIndexNames(),
-            listener
-        );
+            request.offset(),
+            request.after(),
+            request.size(),
+            SnapshotsInProgress.get(state),
+            request.verbose(),
+            request.includeIndexNames()
+        ).getMultipleReposSnapshotInfo(listener);
```

Comment:
I'm inclined to leave it like this for now, if only because it would be a much larger diff to make every field access into a `request.method()` call. Also `GetSnapshotsRequest` includes other junk that shouldn't be relevant here, timeouts and task IDs and so on. Maybe we'll pull out an object for just the relevant fields in a future change.

I'd also like to keep this as a method object and avoid passing things around between individual methods where they're constant for the entire operation. It's quite a help to refactoring if you don't have to worry about whether this thing called `request` is the same as the `request` you saw in some other method or whether it's been modified slightly on the way through. I expect we'll undo this eventually, once the overall structure makes more sense.